MachObfuscator/Mach/Mach+Replacing.swift

```diff
@@ -25,16 +25,32 @@ private extension String {
 
 private extension Mach {
     mutating func replaceSymbols(withMap map: SymbolManglingMap, imageURL: URL, paths: ObfuscationPaths) {
+        // Obfuscate from more specific to less specific objects
+        // When class name is not overwritten debugging other obfuscations is simpler.
+
+        if let methTypeSection = objcMethTypeSection {
+            data.replaceStrings(inRange: methTypeSection.range.intRange,
+                                withMapping: MethTypeObfuscator(withMap: map).generateObfuscatedMethType(methType:))
+        }
+
         if let methNameSection = objcMethNameSection {
             data.replaceStrings(inRange: methNameSection.range.intRange, withMapping: map.selectors)
         }
-        if let classNameSection = objcClassNameSection {
-            data.replaceStrings(inRange: classNameSection.range.intRange, withMapping: map.classNames)
+
+        // Obfuscate property types
+        objcClasses.flatMap { $0.properties }.forEach { property in
+            var attrs = property.attributeValues
+            let typename = property.typeAttribute
+            let newTypename = MethTypeObfuscator(withMap: map).generateObfuscatedMethType(methType: typename)
+            if newTypename != typename {
+                attrs[0] = newTypename
```

Comment:
What do you think about creating a computed setter for `typeAttribute`?

Reply:
I was thinking of if but I decided it is not worth of it. It would be probably used only in one place (here), but the second thing is more important: `ObjcProperty` provides read-only view of the metadata and is implemented by `struct`. To update `typeAttribute` one has to have mutable `Mach` instance to invoke `replaceRangeWithPadding`. This is not a straight-forward operation so I think that is better (or enough) that is performed in a bit more complicated way.

Reply:
Right. That is ok for me!